### ISSUE TITLE
1.9.5 release notes. (561)

### ISSUE THREAD

**Issue by GalacticEmperor1** [pygame-community/pygame-ce#478]
#  **Issue &numero;561 opened by *[illume](https://github.com/illume)* at 2018-10-16 07:31:18**

Need to gather together the release notes like in the previous release.

- [x] should be in markdown format(for the github release notes), which can be converted to html (for the website).
- [x] make a list of people to thank for the credits section.
- [x] link to all the PRs.
- [x] make a summary including important or interesting features/fixes.

Full pygame 1.9.5 release checklist here: https://github.com/pygame/pygame/issues/495<hr>

#  Comments

# # #  **[illume](https://github.com/illume)* commented at 2018-10-16 07:36:45*

ED: ok, here is the draft up to date as of 2019/3/31.

---

<p align="center"><a href="https://www.pygame.org/wiki/about"><img src="https://www.pygame.org/images/logo.png" alt="pygame 1.9.5" width="300px"/></a></p>

# #  TLDR; Some highlights.

Every single source file has been heavily modified and moved in this release. Initial (source code only) support for SDL2 has been merged in. We also support compiling with SDL1 in the same code base, so the migration to pygame 2 is easier. pygame 2 will be released with SDL2 being the default backend when some remaining issues are ironed out. The 1.9.x releases will continue with SDL1 until then. Also, the C API of pygame is undergoing a transformation with lots of cleanups. Then there have been plenty of other cleanups all throughout the python code as well. There's still lots to clean up, but things should be significantly easier for people to [contribute](https://www.pygame.org/contribute.html) (👋 hello and thanks new contributors!). The documentation has been improved with better examples links, search functionality, and improved navigation. Support for older Macs, and newer Macs has been improved. The mask, midi, draw, and math modules have gotten lots of polish with rough edges removed.

# #  Thanks

A very special thanks to the people who have volunteered commits to pygame since the last release. In alphabetical order...

(@alphaCTzo7G) | Amos Bastian (@amosbastian) | Andrey Lopukhov (@andreyx86) | Augusto Noronha (@augusto2112) | Bernardo Sulzbach (@bernardosulzbach) | (@Bottersnike) | Cai Q.T. (@CAIQT) | (@Cerealdragon) | Charles (@charlesej) | (@ChrisWeissNike) | (@cmtrapp02) | Daniel Molina (@daniel-molina) | David Caughell (@DavidCaughell) | David Lönnhager (@dlon) | (@dr0id) | burmer (@dtschirmer) | (@e1000) | xFly_Dragon (@husano896) | (@IchMageBaume) | René Dudfield (@illume) | (@LambdaGoblin) | Lenard Lindstrom (@llindstrom) | François Magimel (@Linkid) | (@LiquidFenrir) | Mark Hurley (@markph0204) | Marius Gedminas (@mgedmin) | (@metulburr) | Michael Farrell (@micolous) | Dominik George (@Natureshadow) | Nik (@nikolas) | Nunu-Willump (@Nunu-Willump) | (@pleboulanger) | Rebecca Chen (@rchen152) | (@robertpfeiffer) | Sett (@sw00) |

Quite a few new people have helped out with this release 🤗

An equally special thanks to everyone else who has contributed in other ways. Thanks to claudeb, imallet, and deshipu for moderating the discord chat to keep it friendly. Thanks to the twitter folk, and the stackoverflow Q&A people. Thanks to everyone who puts their game up on the pygame website for others to look at. Thanks to people making tutorials and sharing articles & books they've written. Thanks to the people working on stuntcat(the community game), and for those who have supported it financially. Thanks to the r/pygame mods and community. Thanks to pyladies groups for running fun sessions, and for making things better. Thanks to the teachers providing feedback, and for doing something fun with their students. Thanks to Debian, pyweek, ludumdare, Raspberrypi, and all the other community groups. 🖤

# #  More details.

https://github.com/pygame/pygame/pull/945 Vector: Explain slices and swizzling
https://github.com/pygame/pygame/pull/944 Update surface documentation
https://github.com/pygame/pygame/pull/937 Make mask offset boundary checking consistent
https://github.com/pygame/pygame/pull/936 Update midi documentation
https://github.com/pygame/pygame/pull/931 Fix drawing 1 pixel wide off-surface ellipses
https://github.com/pygame/pygame/pull/929 Use SDL_RWops instead of pg_FopenUTF8.
https://github.com/pygame/pygame/pull/928 Fix LayeredDirty's inconsistent use of its source rect
https://github.com/pygame/pygame/pull/927 Workaround for locale errors with PyPy
https://github.com/pygame/pygame/pull/926 Update dependencies for manylinux builds.
https://github.com/pygame/pygame/pull/924 Upload .whl binaries to a github draft release when creating tag.
https://github.com/pygame/pygame/pull/923 Update scrap documentation
https://github.com/pygame/pygame/pull/922 https://github.com/pygame/pygame/pull/881 https://github.com/pygame/pygame/pull/879 Fix zero-sized mask segfaults
https://github.com/pygame/pygame/pull/920 Fix the SystemError when no type specific clipboard content exists.
https://github.com/pygame/pygame/pull/917 Update scrap tests
https://github.com/pygame/pygame/pull/914 Update freetype tests
https://github.com/pygame/pygame/pull/911 https://github.com/pygame/pygame/pull/893 https://github.com/pygame/pygame/pull/889 https://github.com/pygame/pygame/pull/876 https://github.com/pygame/pygame/pull/874 https://github.com/pygame/pygame/pull/844 https://github.com/pygame/pygame/pull/840 https://github.com/pygame/pygame/pull/831 https://github.com/pygame/pygame/pull/819 Improved mask tests
https://github.com/pygame/pygame/pull/900 Add sprite tests
https://github.com/pygame/pygame/pull/899 bugfix for dirty sprite when using a source rect
https://github.com/pygame/pygame/pull/891 Add missing Vector2.magnitude()
https://github.com/pygame/pygame/pull/873 Fix mask index out of bounds segmentation fault
https://github.com/pygame/pygame/pull/806 Mask constructor 'fill' argument
https://github.com/pygame/pygame/pull/807 Use version directives consistently in documentation and update style
https://github.com/pygame/pygame/pull/805 Event functions: pump parameter and keyword arguments
https://github.com/pygame/pygame/pull/803 Add customized repr()/str() output for Mask objects
https://github.com/pygame/pygame/pull/802 Fix aaline()/aalines() not drawing on a surface's border
https://github.com/pygame/pygame/pull/797 Keep surface alpha value for copied surfaces with SRCALPHA flag
https://github.com/pygame/pygame/pull/760 https://github.com/pygame/pygame/pull/707 Load images on multiple threads properly
https://github.com/pygame/pygame/pull/783 Fix overlap_mask() making incorrect mask
https://github.com/pygame/pygame/pull/751 https://github.com/pygame/pygame/pull/749 https://github.com/pygame/pygame/pull/748 https://github.com/pygame/pygame/pull/746 https://github.com/pygame/pygame/pull/745 Added get_init() to all modules with an init() function
https://github.com/pygame/pygame/pull/725 Allow camera module to be used on Windows
https://github.com/pygame/pygame/pull/722 API version macros
https://github.com/pygame/pygame/pull/698 Use pre styles in docs
https://github.com/pygame/pygame/pull/670 Converting between ANSI notes and MIDI notes or frequencies
https://github.com/pygame/pygame/pull/663 Compile for PyPy 3 and PyPy fixes
https://github.com/pygame/pygame/pull/665 Chimp tutorial: reindent and clean some code
https://github.com/pygame/pygame/pull/664 Chimp tutorial: fix reST syntax
https://github.com/pygame/pygame/pull/656 https://github.com/pygame/pygame/pull/649 Handle Unicode objects and paths properly
https://github.com/pygame/pygame/pull/631 Fix failing tests test_aapolygon and test_pie
https://github.com/pygame/pygame/pull/630 Fix freetype rotation rendering
https://github.com/pygame/pygame/pull/596 aaline cleanup
https://github.com/pygame/pygame/pull/582 draw.aaline: blend correctly
https://github.com/pygame/pygame/pull/577 https://github.com/pygame/pygame/pull/573 Fixed FreeType memory leaks
https://github.com/pygame/pygame/pull/571 Fix surface.blits() bugs

https://github.com/pygame/pygame/pull/560 Added math.Vector2 subclass test for issue
https://github.com/pygame/pygame/pull/558 Raise TypeError on invalid point arguments
https://github.com/pygame/pygame/pull/557 Update typecheck to allow subtypes
https://github.com/pygame/pygame/pull/545 update system font lists in sysfont.py
https://github.com/pygame/pygame/pull/538 Mac fonts support using system_profiler
https://github.com/pygame/pygame/pull/559 Allow clean silencing of support prompt
https://github.com/pygame/pygame/pull/553 Fix homebrew travis mac builds
https://github.com/pygame/pygame/pull/551 Fixed freezing while playing music from file objects
https://github.com/pygame/pygame/pull/550 updated for python 3
https://github.com/pygame/pygame/pull/546 Improve message on failed imports & cleanup
https://github.com/pygame/pygame/pull/544 Spelling fix in examples/glcube.py
https://github.com/pygame/pygame/pull/540 DirtySprite subclass allow other _layer values
https://github.com/pygame/pygame/pull/536 dependency version into README & Cleanup SDL1.2
https://github.com/pygame/pygame/pull/534 More test cleanups.
https://github.com/pygame/pygame/pull/535 draw_py for Python algos & aaline testing
https://github.com/pygame/pygame/pull/534 More Test cleanup
https://github.com/pygame/pygame/pull/518 Organise the docs modules by usage, add search form.
https://github.com/pygame/pygame/pull/517 Test cleanup n+1 : yet another test cleanup branch
https://github.com/pygame/pygame/pull/516 PY_VERSION_HEX < 0x02070000 : cleanup
https://github.com/pygame/pygame/pull/515 Fix draw polygon ; tests & code cleanup
https://github.com/pygame/pygame/pull/514 Cleanup Python < 2.7 references
https://github.com/pygame/pygame/pull/511 Fix odd-width ellipses not drawing correctly
https://github.com/pygame/pygame/pull/510 Update unit tests draw.ellipse, add unit tests draw.(aa)line(s) 
https://github.com/pygame/pygame/pull/509 Midi tests : add, refactor, cleanup
https://github.com/pygame/pygame/pull/508 More test method cleanup
https://github.com/pygame/pygame/pull/507 Improve mac compile times on travis.
https://github.com/pygame/pygame/pull/505 Remove "if 1:" pattern in tests
https://github.com/pygame/pygame/pull/504 Cleanup mixer test
https://github.com/pygame/pygame/pull/503 Clang format all the src_c/.c files
https://github.com/pygame/pygame/pull/502 Cleanup some test methods & test/sndarray_test.py
https://github.com/pygame/pygame/pull/501 Add draw_ellipse unit test
https://github.com/pygame/pygame/pull/500 Test for overlap mask 
https://github.com/pygame/pygame/pull/499 Remove boilerplate imports from top of test files
https://github.com/pygame/pygame/pull/498 Adding a test for Thick Line Bug
https://github.com/pygame/pygame/pull/493 0 sized masks are now possible
https://github.com/pygame/pygame/pull/489 transform.scale does not crash for zero sized surfaces
https://github.com/pygame/pygame/pull/488 Added unit test 'test_zero_surface_transform' for issue 411
https://github.com/pygame/pygame/pull/486 Moved jquery.plugin.docscomments.js into pygameweb.
https://github.com/pygame/pygame/pull/485 Repo cleanup. 8 things in top level. Rest in buildconfig/. Old stuff removed.
https://github.com/pygame/pygame/pull/484 Added mac sdl2 support. Changed config.py -sdl2
https://github.com/pygame/pygame/pull/483 Support older mac core duo 2 cpus
https://github.com/pygame/pygame/pull/482  The giant SDL2_patches merging PR.
https://github.com/pygame/pygame/pull/476 Do not try to add portmidi on freebsd. It does not exist.
https://github.com/pygame/pygame/pull/475 Resurrect Python 3.4 builds on Appveyor

```
<hr>

# # #  **[dlon](https://github.com/dlon)* commented at 2019-03-23 14:44:34*

Maybe some of these could be added:

# 922 # 881 # 879 Fix zero-sized mask segfaults
# 920 Fix the SystemError when no type specific clipboard content exists.
# 917 Update scrap tests
# 914 Update freetype tests
More test cleanups.
# 911 # 893 # 889 # 876 # 874 # 844 # 840 # 831 # 819 Improved mask tests
# 900 Add sprite tests
# 899 bugfix for dirty sprite when using a source rect
# 891 Add missing Vector2.magnitude()
# 873 Fix mask index out of bounds segmentation fault
# 806 Mask constructor 'fill' argument
# 807 Use version directives consistently in documentation and update style
# 805 Event functions: pump parameter and keyword arguments
# 803 Add customized repr()/str() output for Mask objects
# 802 Fix aaline()/aalines() not drawing on a surface's border
# 797 Keep surface alpha value for copied surfaces with SRCALPHA flag
# 760 # 707 Load images on multiple threads properly
# 783 Fix overlap_mask() making incorrect mask
# 751 # 749 # 748 # 746 # 745 Added get_init() to all modules with an init() function
# 725 Allow camera module to be used on Windows
# 722 API version macros
# 698 Use pre styles in docs
# 670 Converting between ANSI notes and MIDI notes or frequencies
# 663 Compile for PyPy 3 and PyPy fixes
# 656 # 649 Handle Unicode objects and paths properly
# 631 Fix failing tests test_aapolygon and test_pie
# 630 Fix freetype rotation rendering
# 596 aaline cleanup
# 582 draw.aaline: blend correctly
# 577 # 573 Fixed FreeType memory leaks
# 571 Fix surface.blits() bugs

I may have overlooked something.
<hr>

# # #  **[illume](https://github.com/illume)* commented at 2019-03-24 16:51:40*

The new contributors need to be added from pr 571 onwards.

Also, there's been a lot of changes since I wrote the release summary.
<hr>

# # #  **[dlon](https://github.com/dlon)* commented at 2019-03-24 17:39:54*

I added the new contributors
